atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/config/OracleConnectionConfig.java

```diff
@@ -0,0 +1,171 @@
+/**
+ * Copyright 2015 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.nexus.db.pool.config;
+
+import static com.palantir.nexus.db.pool.config.ConnectionProtocol.TCP;
+import static com.palantir.nexus.db.pool.config.ConnectionProtocol.TCPS;
+
+import java.io.File;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.palantir.nexus.db.DBType;
+
+@JsonDeserialize(as = ImmutableOracleConnectionConfig.class)
+@JsonSerialize(as = ImmutableOracleConnectionConfig.class)
+@JsonTypeName(OracleConnectionConfig.TYPE)
+@Value.Immutable
+public abstract class OracleConnectionConfig extends ConnectionConfig {
+
+    public static final String TYPE = "oracle";
+
+    public abstract String getHost();
+    public abstract int getPort();
+
+    @Override
+    @Value.Derived
+    public String getUrl() {
+        if (getServerDn().isPresent()) {
+            return String.format("jdbc:oracle:thin:@(DESCRIPTION=" +
+                            "(ADDRESS=(PROTOCOL=%s)(HOST=%s)(PORT=%s))" +
+                            "(CONNECT_DATA=(SID=%s))" +
+                            "(SECURITY=(SSL_SERVER_CERT_DN=\"%s\")))",
+                    getProtocol().getUrlString(), getHost(), getPort(), getSid(), getServerDn().get());
+        } else {
+            return String.format("jdbc:oracle:thin:@(DESCRIPTION=" +
+                            "(ADDRESS=(PROTOCOL=%s)(HOST=%s)(PORT=%s))" +
+                            "(CONNECT_DATA=(SID=%s)))",
+                    getProtocol().getUrlString(), getHost(), getPort(), getSid());
+        }
+    }
+
+    @Override
+    @Value.Default
+    public String getDriverClass() {
+        return "oracle.jdbc.driver.OracleDriver";
+    }
+
+    @Override
+    @Value.Default
+    public String getTestQuery() {
+        return "SELECT 1 FROM dual";
+    }
+
+    public abstract String getSid();
+
+    public abstract Optional<String> getServerDn();
+
+    @Value.Default
+    public boolean getMatchServerDn() {
+        return getServerDn().isPresent();
+    }
+
+    @Value.Default
+    public boolean getTwoWaySsl() {
+        return false;
```

Comment:
So I know this used to be false, but given the infosec push I'm pretty sure we should be setting it to true, even if we think that's breaking.